lib/http/request.js

```diff
@@ -54,6 +57,35 @@ class HttpRequest extends EventEmitter {
     __globalSettings__ = val;
   }
 
+  static resetHttpKeepAliveAgents() {
+    __httpKeepAliveAgent__ = null;
+    __httpsKeepAliveAgent__ = null;
+  }
+
+  static getHttpKeepAliveAgent(keepAliveMsecs, maxSockets) {
```

Comment:
can you refactor this into a common method and call the `https` something like "getSecureKeepAliveAgent"?

Also try to use an object literal for function arguments.